examples/teams_matrix-multiply.cpp

```diff
@@ -0,0 +1,928 @@
+  //
+   RAJA::expt::launch<launch_policy>(RAJA::expt::DEVICE,
+    RAJA::expt::Resources(RAJA::expt::Teams(N),
+                          RAJA::expt::Threads(N)),
+        [=] RAJA_HOST_DEVICE(RAJA::expt::LaunchContext ctx) {
+
+     RAJA::expt::loop<teams_x>(ctx, col_range, [&] (int col) {
+       RAJA::expt::loop<threads_x>(ctx, row_range, [&] (int row) {
+
+            double dot = 0.0;
+            for (int k = 0; k < N; ++k) {
+              dot += d_Aview(row, k) * d_Bview(k, col);
+            }
+
+            d_Cview(row, col) = dot;
+
+        });
+     });
+  });
+
+  hipErrchk(hipMemcpy( C, d_C, N * N * sizeof(double), hipMemcpyDeviceToHost ));
+  checkResult<double>(Cview, N);
+//printResult<double>(Cview, N);
+
+
+//----------------------------------------------------------------------------//
+
+  std::cout << "\n Running HIP tiled mat-mult (RAJA-POL5)...\n";
+
+  std::memset(C, 0, N*N * sizeof(double));
+  hipErrchk(hipMemcpy( d_C, C, N * N * sizeof(double), hipMemcpyHostToDevice ));
+
+  //
+  // This policy collapses the col and row loops into a single HIP kernel
+  // using two-dimensional HIP thread blocks with x and y dimensions defined
+  // by HIP_BLOCK_SIZE arguments.
+  //
+  // When the matrix dimension N is an integer multiple of HIP_BLOCK_SIZE,
+  // the HIP grid and thread dimension kernel launch parameters will be the
+  // same as in this kernel and the one above.
+  //
+  RAJA::expt::launch<launch_policy>(RAJA::expt::DEVICE,
+    RAJA::expt::Resources(RAJA::expt::Teams(NTeams,NTeams),
+                          RAJA::expt::Threads(GPU_TB_SZ,GPU_TB_SZ)),
+        [=] RAJA_HOST_DEVICE(RAJA::expt::LaunchContext ctx) {
+
+        RAJA::expt::tile<teams_y>
+          (ctx, GPU_TB_SZ, row_range, [&] (RAJA::RangeSegment const &x_tile) {
+            RAJA::expt::tile<teams_x>
+              (ctx, GPU_TB_SZ, col_range, [&] (RAJA::RangeSegment const &y_tile) {
+
+                RAJA::expt::loop<threads_y>(ctx, y_tile, [&] (int col) {
+                    RAJA::expt::loop<threads_x>(ctx, x_tile, [&] (int row) {
+
+                        double dot = 0.0;
+                        for (int k = 0; k < N; ++k) {
+                          dot += d_Aview(row, k) * d_Bview(k, col);
+                        }
+
+                        d_Cview(row, col) = dot;
+                      });
+                 });
+
+            });
+       });
+
+   });
+  hipErrchk(hipMemcpy( C, d_C, N * N * sizeof(double), hipMemcpyDeviceToHost ));
+  checkResult<double>(Cview, N);
+//printResult<double>(Cview, N);
+#endif // if RAJA_ENABLE_HIP
+
+//----------------------------------------------------------------------------//
+#if defined(RAJA_ENABLE_CUDA)
+
+  std::cout << "\n Running CUDA tiled mat-mult with shared memory ...\n";
+
+  std::memset(C, 0, N*N * sizeof(double));
+
+  using seq_loop =  RAJA::expt::LoopPolicy<RAJA::loop_exec, RAJA::loop_exec>;
+
+  RAJA::expt::launch<launch_policy>(RAJA::expt::DEVICE,
```

Comment:
Hi folks, I've pushed this teams_matrix-multiply.cpp example that presents a number of variants of matrix-multiply. This includes Tiling with shared memory. I'm still working on the documentation, but I think looks pretty cool! I also have a native CUDA version on top.